src/ni/vsbuild/Pipeline.groovy

```diff
@@ -88,8 +88,15 @@ class Pipeline implements Serializable {
          // need to bind the variable before the closure - can't do 'for (version in lvVersions)'
          def lvVersion = version
 
+         String fullLabel
+         if ("${pipelineInformation.nodeLabel}" && !"${pipelineInformation.nodeLabel}".allWhitespace){
```

Comment:
This will not work correctly if `pipelineInformation.nodeLabel == null`. The string interpolation to a GString will give the result a value of `"null"` (as a non-empty string). Don't interpolate here. Also, an empty string in groovy evaluates to false, so trim() will work to determine if we have only whitespace.

```
String nodeLabel = lvVersion
if(pipelineInformation.nodeLabel?.trim()) {
   nodeLabel = "$nodeLabel && pipelineInformation.nodeLabel"
}
...
script.node(nodeLabel) {...}
```

Reply:
Updated!